core/pkg/artifacts/manifest_test.go

```diff
@@ -0,0 +1,192 @@
+package artifacts
+
+import (
+	"bufio"
+	"compress/gzip"
+	"encoding/json"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/wandb/wandb/core/pkg/service"
+)
+
+func TestNewManifestFromProto(t *testing.T) {
+	proto := &service.ArtifactManifest{
+		Version:       1,
+		StoragePolicy: "policy",
+		Contents: []*service.ArtifactManifestEntry{
+			{
+				Path:   "path1",
+				Digest: "digest1",
+				Size:   123,
+				Extra: []*service.ExtraItem{
+					{Key: "key1", ValueJson: `"value1"`},
+				},
+			},
+		},
+	}
+
+	manifest, err := NewManifestFromProto(proto)
+	assert.NoError(t, err)
+	assert.Equal(t, int32(1), manifest.Version)
```

Comment:
nit: would prefer something like below instead of all the hardcoding instead for testing
```suggestion
	assert.Equal(t, int32(proto.Version), manifest.Version) // something like that
```